google/cloud/bigtable/tools/generate_rpc_policy_defaults.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Generate header file for retry policy defaults from GAPIC yamls
+
+GAPIC suggests default values for clients to design their retry policy.
+
+For Bigtable admin APIs it is defined in bigtableadmin_gapic.yaml
+https://raw.githubusercontent.com/googleapis/googleapis/master/google/bigtable/admin/v2/bigtableadmin_gapic.yaml
+
+For Bigtable data APIs it is defined in bigtable_gapic.yaml
+https://raw.githubusercontent.com/googleapis/googleapis/master/google/bigtable/v2/bigtable_gapic.yaml
+
+This script reads both yamls and generates a C++ header file that define
+constants which is used in retry and backoff policies for Bigtable Clients.
+
+Usage:
+
+    python ../tools/generate_rpc_policy_defaults.py \
+    | clang-format > rpc_policy_defaults.h
+"""
+
+import yaml
+import sys
+import urllib
+
+FILE_HEADER = """
```

Comment:
We probably want all of this in a static file, take a look at:

https://github.com/GoogleCloudPlatform/google-cloud-cpp/blob/master/google/cloud/bigtable/internal/readrowsparser_acceptance_tests.inc

All the code that does not need to be generated is in a normal file, which includes the generated code:

https://github.com/GoogleCloudPlatform/google-cloud-cpp/blob/c313463e06cf385525b9cd0eeb0d2467bf185e7c/google/cloud/bigtable/internal/readrowsparser_test.cc#L251

Could you do that here to make the generator (and the generated code) smaller?

Reply:
fixed.